File: src/engine/commands/gtalkcommand.cpp
#include "gtalkcommand.h"

#include "util.h"


#define super Command

GtalkCommand::GtalkCommand(QObject *parent) :
    super(parent) {

    setDescription("Talk to everyone who's in the same group.\n"
                   "\n"
                   "Example: gtalk Where do we go now?");
}

GtalkCommand::~GtalkCommand() {
}

void GtalkCommand::execute(Player *player, const QString &command) {

    super::execute(player, command);

    if (!assertWordsLeft("Say what?")) {
        return;
    }

    if (player->group().isNull()) {
        send("You are not in a group.");
    } else {
        QString message = takeRest();
        QString text = (message.endsWith(".") || message.endsWith("?") || message.endsWith("!")) ?
                       "(Group) %1 mentioned, \"%2\"" : "(Group) %1 mentioned, \"%2.\"";

        player->group()->send(text.arg(player->name(), message));
    }
}

File: data/commands/gtalkcommand.js
function GtalkCommand() {
    Command.call(this);

    this.setDescription("Talk to everyone who's in the same group.\n" +
                        "\n" +
                        "Example: gtalk Where do we go now?");
}

GtalkCommand.prototype = new Command();
GtalkCommand.prototype.constructor = GtalkCommand;

GtalkCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    if (!this.assertWordsLeft("Say what?")) {
        return;
    }

    if (player.group) {
        var message = this.takeRest();
        var text = (message.endsWith(".") || message.endsWith("?") || message.endsWith("!")) ?
                   "(Group) %1 mentioned, \"%2\"" : "(Group) %1 mentioned, \"%2.\"";

        player.group.send(text.arg(player.name, message));
    } else {
        this.send("You are not in a group.");
    }
}

CommandRegistry.registerCommand("gtalk", new GtalkCommand());
